### PR TITLE
fix(jax-backend): set `symmetrize_input=False` in `jax.eigh` for consistency

### LINKED ISSUE #23098: [Bug]:  `jax.eigh` Returns Inconsistent Eigenvalues Compared to Other Frameworks Due to Implicitly Passing `symmetrize_input=True`
### Bug Explanation

Currently,we're calling  [jax.numpy.linalg.eigh](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.linalg.eigh.html) with  arguments `jax.numpy.linalg.eigh(a, UPLO=None, symmetrize_input=True)` (default) although at ivy's `jax backend` we didn't specify `symmetrize_input=True` as we can see in-here ->[LINK](https://github.com/unifyai/ivy/blob/main/ivy/functional/backends/jax/linear_algebra.py#L116) .i.e. we are implicitly setting `symmetrize_input=True`

what `symmetrize_input=True` does is If set to `True`, the matrix is symmetrized before the
      eigendecomposition by computing $\frac{1}{2}(x + x^H)$  
refs: [jax.lax.linalg.eigh](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.linalg.eigh.html) this what is used by `jax.numpy.linalg.eigh` at backend.

### Steps to Reproduce Bug

Current Ivy definition of `jax.numpy.linalg.eigh` i.e. `symmetrize_input=True` :

```py
# path: ivy/functional/backends/jax/linear_algebra.py
@with_unsupported_dtypes(
    {"0.4.14 and below": ("bfloat16", "float16", "complex")},
    backend_version,
)
def eigh(
    x: JaxArray, /, *, UPLO: str = "L", out: Optional[JaxArray] = None
) -> Tuple[JaxArray]:
    result_tuple = NamedTuple(
        "eigh", [("eigenvalues", JaxArray), ("eigenvectors", JaxArray)]
    )
    eigenvalues, eigenvectors = jnp.linalg.eigh(x, UPLO=UPLO) # ==  jnp.linalg.eigh(x, UPLO=UPLO, symmetrize_input=True)
    return result_tuple(eigenvalues, eigenvectors)

```
Thus, when we call it and test with the same input for other frameworks supported by ivy:
```py
import ivy
import warnings 
jax.config.update('jax_enable_x64', True)


# tensorflow will give error due to me calling  `tolist` for better formatting 
frame = ["numpy", "jax", "paddle", "tensorflow", "torch"]

with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    for framework in frame:
        ivy.set_backend(framework)
        try:
            x = ivy.array([[1., 1.], [2., 1.]],dtype=ivy.float64)
            eigenvalues, eigenvectors = ivy.eigh(x)
            
            # Convert arrays to native Python lists if necessary
            if not isinstance(eigenvalues, list):
                eigenvalues = eigenvalues.tolist()
            if not isinstance(eigenvectors, list):
                eigenvectors = eigenvectors.tolist()
            
            formatted_eigenvalues = ', '.join([f"{val.real:.5f}+{val.imag:.5f}j" for val in eigenvalues])
            formatted_eigenvectors = ', '.join([f"{val.real:.5f}+{val.imag:.5f}j" for row in eigenvectors for val in row])

            print(f"Framework: {framework}")
            print(f"  Eigenvalues: [{formatted_eigenvalues}]")
            print(f"  Eigenvectors: [{formatted_eigenvectors}]")
            print('-'*50)

        except Exception as e:
            print(f"Framework: {framework}")
            print(f"  Error: {e}")
            print('-'*50)

"""
Framework: numpy
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
Framework: jax
  Eigenvalues: [-0.50000+0.00000j, 2.50000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
Framework: paddle
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
Framework: tensorflow
  Error: EagerTensor object has no attribute 'tolist'. 
        If you are looking for numpy-related methods, please run the following:
        from tensorflow.python.ops.numpy_ops import np_config
        np_config.enable_numpy_behavior()
      
--------------------------------------------------
Framework: torch
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------

# TF
ivy.set_backend("tensorflow")
x = ivy.array([[1., 1.], [2., 1.]],dtype=ivy.float64)
y = ivy.eigh(x)
print(y)
eigh(eigenvalues=ivy.array([-1.,  3.]), eigenvectors=ivy.array([[-0.70710678,  0.70710678],
       [ 0.70710678,  0.70710678]]))
"""

```

when I set `symmetrize_input=False`:

```py
Framework: numpy
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
Framework: jax
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
Framework: paddle
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
Framework: torch
  Eigenvalues: [-1.00000+0.00000j, 3.00000+0.00000j]
  Eigenvectors: [-0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j, 0.70711+0.00000j]
--------------------------------------------------
```

### Environment

linux+docker+vscode

### Ivy Version

v1.1.9

### Backend

- [ ] NumPy
- [ ] TensorFlow
- [ ] PyTorch
- [X] JAX

### Device

CPU

### PR DESCRIPTION
**Pull request by akshatvishu**
Set the `symmetrize_input` argument to `False` in `jax.eigh` to ensure consistent behavior with other backends. The default behavior in `jax` was leading to different eigenvalues than other frameworks.

## Related Issue 

Close #23098

### CODE REVIEW

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi @akshatvishu,
Your code looks sensible to me, but I guess we have a more systematic issue here. In the main branch the failing tests for this function are `torch` and `numpy`, not `jax`. That mean we both have a bug in those backends as well as a potential bug on the test (according to #23098).

Can you fix those issues and make sure `eigh` passes everything before we proceed? Otherwise I feel we would be digging our own hole to fall in

**Comment by KareemMAX** [comment]
Also, I found out that the test is quite non-standard (and a bit old). [ref](https://github.com/unifyai/ivy/blob/main/ivy_tests/test_ivy/test_functional/test_core/test_linalg.py#L479)

You might want to ask for @sherry30's opinion. Or maybe reach out to our `#testing` channel on discord.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
